gTea/app/src/main/java/com/nechaieva/gtea/OrderProcessor.java

```diff
@@ -0,0 +1,64 @@
+package com.nechaieva.gtea;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class OrderProcessor {
+
+    List<String> menu;
+    final int SENSITIVITY = 3;
+    public final String NO_CANDIDATE = "";
+
+    OrderProcessor(String[] menu) {
+        this.menu = Arrays.asList(menu);
+    }
+
+    OrderProcessor(ArrayList<String> menu) {
+        this.menu = menu;
+    }
+
+    public String findInMenu(String query) {
+        Match match = findClosest(query);
+        return match.getDistance() < SENSITIVITY ? match.getMatchString() : NO_CANDIDATE;
+    }
+
+    private Match findClosest(String query) {
+        // Probably a better approach than a straightforward iterate-through exists
```

Comment:
As you have to compute the distance for all elements in the list at least once, you have to iterate through it at least once in the worst case. You could consider adding optimisation heuristics e.g. breaking early if you have a zero distance match.

Reply:
Nice idea. Breaking in case there was already a better match would be possible as well ;-)

I like this Knuth citation about optimization